[CI] CoreWithMultipleProjectsClientYamlTestSuiteIT class failing

**Build Scans:**
- [elasticsearch-periodic #6810 / openjdk24_entitlements_true_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/xabxvl6loehcc)
- [elasticsearch-pull-request #60847 / part-2](https://gradle-enterprise.elastic.co/s/f6o7jxq5q4h26)
- [elasticsearch-pull-request #60840 / part-2](https://gradle-enterprise.elastic.co/s/agw6wbeeiucfg)
- [elasticsearch-pull-request #60827 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/aefgs347vqx6e)
- [elasticsearch-periodic #6789 / openjdk21_entitlements_true_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/hnoqpfiaww3ua)
- [elasticsearch-pull-request #60506 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/wcwwccudtvkto)
- [elasticsearch-pull-request #60458 / part-2](https://gradle-enterprise.elastic.co/s/ov6wvelvpmq2q)
- [elasticsearch-pull-request #60388 / part-2](https://gradle-enterprise.elastic.co/s/ajymw66y57dzs)
- [elasticsearch-pull-request #60204 / part-2](https://gradle-enterprise.elastic.co/s/yrqdgy5bc3wce)
- [elasticsearch-periodic #6761 / openjdk21_checkpart2_java-fips-matrix](https://gradle-enterprise.elastic.co/s/yvhlekltkho6i)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:multi-project:core-rest-tests-with-multiple-projects:yamlRestTest" --tests "org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT.test {yaml=reindex/20_validation/max_docs shoule be greater than slices}" -Dtests.seed=21E9319448E74A01 -Dtests.jvm.argline="-Des.entitlements.enabled=true" -Dtests.locale=nnh-CM -Dtests.timezone=Antarctica/Rothera -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [reindex/20_validation:146]: expected [2xx] status code but api [index] returned [503 Service Unavailable] [{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];","stack_trace":"org.elasticsearch.cluster.block.ClusterBlockException: blocked by: [SERVICE_UNAVAILABLE/2/no master];\n"}],"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];","stack_trace":"org.elasticsearch.cluster.block.ClusterBlockException: blocked by: [SERVICE_UNAVAILABLE/2/no master];\n"},"status":503}]
```

**Issue Reasons:**
- [main] 3 consecutive failures in step checkpart2_release-tests
- [main] 13 failures in class org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT (1.4% fail rate in 942 executions)
- [main] 7 failures in step part-2 (2.0% fail rate in 354 executions)
- [main] 3 failures in step checkpart2_release-tests (100.0% fail rate in 3 executions)
- [main] 3 failures in pipeline elasticsearch-periodic (17.6% fail rate in 17 executions)
- [main] 10 failures in pipeline elasticsearch-pull-request (2.8% fail rate in 357 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java
+++ b/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java
@@ -1366,11 +1366,12 @@ public class IndexShard extends AbstractIndexShardComponent implements IndicesCl
     }
 
     public FlushStats flushStats() {
+        final Engine engine = getEngineOrNull();
         return new FlushStats(
             flushMetric.count(),
             periodicFlushMetric.count(),
             TimeUnit.NANOSECONDS.toMillis(flushMetric.sum()),
-            getEngineOrNull() != null ? getEngineOrNull().getTotalFlushTimeExcludingWaitingOnLockInMillis() : 0L
+            engine != null ? engine.getTotalFlushTimeExcludingWaitingOnLockInMillis() : 0L
         );
     }
 

